specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json

```diff
@@ -1611,6 +1660,49 @@
           "description": "Error message, if any, returned when deploying the resource."
         }
       }
+    },
+    "ServicePrincipal": {
+      "description": "The Service Principal Id for the subscription.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "ARM id of the resource."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Name of the resource."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of the resource."
+        },
+        "properties": {
+          "title": "Properties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ServicePrincipalProperties",
+          "description": "The Service Principal properties for the subscription"
+        }
```

Comment:
This needs to be a proxy resource

Reply:
Yes it is a Proxy resource. We have updated in Manifest

Reply:
No like reference a ProxyResource object. For example you can see  how other swaggers reference : https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/common-types/resource-management/v3/types.json#L91

Reply:
Thank you got it! I will fix it now

Reply:
@raosuhas When I reference ProxyResource, I got the below error for [RequiredReadOnlySystemData](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R4009)
Since we are doing get only, Can we return systemData.createdBy as **SYSTEM**

![image](https://user-images.githubusercontent.com/16600949/152625772-b0c12913-6ab2-4abc-bf6a-60740186778f.png)

Reply:
@raosuhas Above comment is fixed and I will resolve once you answer for the above question.